packages/Telephony/Actions/TerminationCharacterBatchInput.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Dialogs;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Components.Telephony.Actions
+{
+    /// <summary>
+    /// Aggregates input until it matches a regex pattern and then stores the result in an output property.
+    /// </summary>
+    public class TerminationCharacterBatchInput : RegexAggregatorInput
+    {
+        /// <summary>
+        /// Class identifier.
+        /// </summary>
+        [JsonProperty("$kind")]
+        public new const string Kind = "Microsoft.Telephony.TerminationCharacterBatchInput";
+
+        private string _terminationCharacter;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TerminationCharacterBatchInput"/> class.
+        /// </summary>
+        /// <param name="regexAggregatorInput">Implementation for aggregatingInput.</param>
+        /// <param name="sourceFilePath">Optional, source file full path.</param>
+        /// <param name="sourceLineNumber">Optional, line number in source file.</param>
+        [JsonConstructor]
+        public TerminationCharacterBatchInput([CallerFilePath] string sourceFilePath = "", [CallerLineNumber] int sourceLineNumber = 0)
+            : base()
+        {
+            // enable instances of this command as debug break point
+            this.RegisterSourceLocation(sourceFilePath, sourceLineNumber);
+        }
+
+        /// <summary>
+        /// Gets or sets the character that will be used to signal that the batch of input is complete.
+        /// </summary>
+        [JsonProperty("terminationCharacter")]
+        public string TerminationCharacter
+        {
+            get
+            {
+                return _terminationCharacter;
+            }
+
+            set
+            {
+                TerminationConditionRegexPattern = value + "$";
+                _terminationCharacter = value;
```

Comment:
Should the termination Character be part of the batch? 

Current flow: 
Bot: Enter your phone number followed by #
User : 1234567#
Bot: You said 1234567#

Might be good to add the expected behavior to our spec

Reply:
= Termination character set to $ accepts any string
= What is the valid set of termination characters that we want to accept?

Reply:
I figured there would be some nice sanitization questions like this one :D
As for including the termination character, I made it default to include it. It should probably be documented.

For something developer facing like this package, I'm inclined to think that sanitization is a dev responsibility (ie. if they pass $ in they want it to accept anything haha). We could just lock down everything to be DTMF specified and not worry about this. Let's have a chat about these details.

Reply:
I'll address sanitization in a future PR.